### DISCUSSION TITLE
PR-curve for one class

### DISCUSSION THREAD

**Discussion by polcaruca** [ultralytics/ultralytics#12904] [Q&A]
Hello,

I have trained a model and created a .pt file through the validation process. 
Currently, I want to plot the PR-curve for only one specific class from among the multiple classes shown 
in the output PR-curve graph. Ideally, I want to hide the other classes so that only one class is visible (and also modify the legend accordingly). 

How can I achieve this?

Thank you.

**Comment by glenn-jocher** [comment]
Hello!

To plot the PR-curve for a specific class and hide others, you can modify the validation script to filter the results for the class of interest before plotting. Here's a simple way to do it using Python:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Validate the model
metrics = model.val()

# Access the PR curve data
pr_data = metrics.pr_curve()

# Filter for a specific class, e.g., class_id = 0
class_id = 0
specific_class_data = pr_data[class_id]

# Now you can plot using specific_class_data
# Use your preferred plotting library, e.g., matplotlib
import matplotlib.pyplot as plt

plt.figure()
plt.plot(specific_class_data['recall'], specific_class_data['precision'])
plt.xlabel('Recall')
plt.ylabel('Precision')
plt.title(f'PR Curve for Class ID {class_id}')
plt.show()
```

This script assumes you're using the Ultralytics YOLO API and have matplotlib installed for plotting. Adjust `class_id` to target the class you're interested in. 

Hope this helps! Let me know if you need further assistance. 😊
